core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -1654,7 +1699,19 @@ public long getCommitLag() {
 
         TableStructureAdapter of(CharSequence tableName, LineTcpParser protoParser) {
             this.tableName = tableName;
-            this.protoParser = protoParser;
+
+            entityNames.clear();
+            entities.clear();
+            for (int i = 0; i < protoParser.getnEntities(); i++) {
```

Comment:
The below code might look somewhat inefficient but it runs only if and when the table is created.